include/components/renderer_interface.hpp

```diff
@@ -33,6 +33,8 @@ class renderer_interface {
    */
   virtual double get_alignment_start(const alignment align) const = 0;
 
+  virtual void apply_tray_position(unique_ptr <polybar::tags::context> &context) = 0;
```

Comment:
```suggestion
  virtual void apply_tray_position(const tags::context& context) = 0;
```

It doesn't really make sense to pass a reference to a unique_ptr, passing a const-reference here is much cleaner (that's what the dispatcher does everywhere else anyway).